### ISSUE TITLE
`sprintf-without-formatters` rule ineffective due to removed `Φ.txt.sprintf` object, failing to match `base='.printf'`

### ISSUE THREAD

**Issue by gemshrine** [objectionary/lints#1357]
`sprintf-without-formatters` selects the same object `wrong-sprintf-arguments` does, and that object no longer exists in EO, so this rule cannot fire either.

`src/main/resources/org/eolang/lints/misc/sprintf-without-formatters.xsl:20`:

```xml
<xsl:for-each select="//o[@base='Φ.txt.sprintf']">
```

There is no `txt` package in `objectionary/eo` any more and no `sprintf` object anywhere in the runtime. Formatting is `eo-runtime/src/main/eo/string/printf.eo`, called as a method on the format string, so what a current XMIR carries is `base=".printf"`:

```
$ grep -rl "txt.sprintf" eo-runtime/target/eo/1-parse/
(nothing)

$ grep -rlo "\.printf" eo-runtime/target/eo/1-parse/*.xmir | wc -l
13
```

The check it performs is still worth having. It reports a format template with none of the supported conversions, on the grounds that formatting without a formatter is pointless — a real mistake that is easy to make and invisible at runtime, since the template simply comes back unchanged.

Adapting it means matching the dispatch rather than the base:

```xml
<xsl:for-each select="//o[@base='.printf']">
```

and reading the template from the receiver, which for a dispatch is emitted as a preceding sibling rather than as a child, so the `$text` extraction on lines 21-38 has to follow it there.

Two things are worth revisiting in the same change. The message on line 60 names `"%s", "%d", "%f", "%x", "%b"` as the supported set, and current `printf` handles more than that — flags, widths and precisions such as `%5d`, `%08.2f`, `%.0f` all work — so a template using only those would now be reported as having no formatters at all. And the message text names `"Φ.txt.sprintf"` three times, which will read oddly once the rule matches something else.

The sibling rule `wrong-sprintf-arguments` has the same dead selector and is reported separately, since the two need different adaptations: that one is about arity and tuple shape, this one about the template's content.

**Comment by 0crat** [comment]
@gemshrine Thanks for reporting a new bug! You've earned +12 points. By reporting bugs, you help our project improve its quality. If you find anything else in the repository that [doesn't look](https://www.yegor256.com/2018/02/06/where-to-find-more-bugs.html) as good as you might expect, [do not hesitate](https://www.yegor256.com/2014/04/13/bugs-are-welcome.html) to [report](https://www.yegor256.com/2018/04/24/right-way-to-report-bugs.html) it. Your running score is [+3614](https://www.eolang.org/zerocracy/objectionary-vitals.html); don't forget to check your [Zerocracy account](https://www.zerocracy.com/) too).
